common/tier4_autoware_utils/include/tier4_autoware_utils/trajectory/trajectory.hpp

```diff
@@ -34,6 +36,27 @@ void validateNonEmpty(const T & points)
   }
 }
 
+template <class T>
+void validateNonSharpAngle(
+  const T & point1, const T & point2, const T & point3, const double angle_threshold = pi / 4)
+{
+  const auto p1 = getPoint(point1);
+  const auto p2 = getPoint(point2);
+  const auto p3 = getPoint(point3);
+
+  const std::vector vec_1to2 = {p2.x - p1.x, p2.y - p1.y, p2.z - p1.z};
+  const std::vector vec_3to2 = {p2.x - p3.x, p2.y - p3.y, p2.z - p3.z};
+  const auto product = std::inner_product(vec_1to2.begin(), vec_1to2.end(), vec_3to2.begin(), 0.0);
+
+  const auto dist_1to2 = calcDistance3d(p1, p2);
+  const auto dist_3to2 = calcDistance3d(p3, p2);
+
+  constexpr double epsilon = 1e-3;
+  if (std::cos(angle_threshold) < product / dist_1to2 / dist_3to2 + epsilon) {
+    throw std::invalid_argument("Invalid sharp angle.");
```

Comment:
@satoshi-ota 
there would be many sharp angle path case and this sharp angle case can be detected by planning error monitor so warn throttle would be better for now.

Reply:
FYI
https://github.com/autowarefoundation/autoware.universe/blob/a0ef0e9f4d9d074aa65dc7a1a08742b4b1a01921/planning/planning_error_monitor/src/planning_error_monitor_node.cpp#L118

Reply:
@satoshi-ota Do you remove this checker function?

Reply:
:thinking: